### ISSUE TITLE
"Error retrieving error string"

### ISSUE THREAD

**Issue by gmartind** [sunsetkookaburra/rust-libesedb#11]
Good morning team! I have updated to the 2.0 package and I'm having problems to open my ESE database: 

```
fn main() {
    let filename = std::env::args()
        .nth(1)
        .unwrap_or("Catalog1.edb".to_string());
    let db = EseDb::open(filename).unwrap();
}
```
Returns the Error:
`thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: Other, error: "Error retrieving error string\nError retrieving error backtrace string" }`

Any info on this?
Thanks for the great work btw!

**Comment by sunsetkookaburra** [comment]
Hello,

It should be fixed now in [libesedb-v0.2.1](https://github.com/sunsetkookaburra/rust-libesedb/commit/987df44c4ff5914bc1f36966ca207718fe6b3fe2).

I don't have tests set up yet so I missed some logic errors when I was refactoring error handling logic. Hopefully I should have some time soon to add that in to avoid something like this in future.

Let me know if this fixes things on your end too and I can close this issue.

\- Oli

**Comment by gmartind** [comment]
Hey, unfortunately it keeps returning the same error :/
![imagen](https://user-images.githubusercontent.com/65733734/236675348-b99be0d7-4843-4338-bd4b-bbf96cec9c3e.png)

**Comment by sunsetkookaburra** [comment]
You bumped the Cargo.toml file to `0.2.1` yep?
The fix worked for me on my system, so I'm a little stumped.

What library function do you call and get the error with? Is it `EseDb::open()` or something else? It's possible there's some deeper issue I'm not catching.

**Comment by gmartind** [comment]
Yep, I'm using 0.2.1.
![imagen](https://user-images.githubusercontent.com/65733734/236688773-c9f52806-e8c1-496c-9fb3-3862ab946006.png)
You were right, seems that the problem now is related to EseDb::table_by_name(...) function.

**Comment by sunsetkookaburra** [comment]
Sorry to re-open it; can you show how you were using it? The "Error retrieving error string" is a bug and means that there are probably some other hidden errors.

**Comment by gmartind** [comment]
Basically, at first I was having a problem with the `EseDb::open()` function, I couldn't open the table that I was able to deal in previous versions. After 0.2.1 it was solved. The problem with the `EseDb::table_by_name(...)` was associated with a typo in the table name.

**Comment by sunsetkookaburra** [comment]
Huh, okay. Thank you though. I'll see if I can reproduce it. Seems likely I've got an issue going between Rust/C strings.

**Comment by sunsetkookaburra** [comment]
Turns out there is an undocumented case in `libesedb_file_get_table_by_utf8_name`. An error return value might not mean the error structure has been set.

Workaround/fix in 2fadffa532639ac60037075eaa97fabf9256f9d1, should be tagged/released shortly.

**Comment by sunsetkookaburra** [comment]
Error message fixed and published in [`0.2.2`](https://crates.io/crates/libesedb/0.2.2)

**Comment by gmartind** [comment]
New problem, now I can't open my tables, although I am sure that they exist and I was able to open them in the past. Getting the error msg `kind: Other, error: "rust-libesedb: Can't find table 'SruDbIdMapTables'"`
